src/cryptography/hazmat/backends/openssl/ciphers.py

```diff
@@ -78,7 +78,8 @@ def __init__(self, backend, cipher, mode, operation):
                 len(iv_nonce), self._backend._ffi.NULL
             )
             self._backend.openssl_assert(res != 0)
-            if operation == self._DECRYPT:
+
+            if operation == self._DECRYPT and mode.tag is not None:
```

Comment:
This branch shouldn't be necessary any more since the tag is being set in `finalize`, correct?